code/modules/recycling/compactor.dm

```diff
@@ -0,0 +1,139 @@
+/obj/machinery/disposal/compactor
+	name = "trash compactor"
+	desc = "A machine used to alleviate recycling problems in the absence of a disposal network."
+	icon_state = "compactor_on" //New sprite indicating fullness?
+	machine_flags = WRENCHMOVE | FIXED2WORK
+	flags = FPRINT
+	req_access = list(access_janitor)
+	template_path = "disposalsbincompactor.tmpl"
+
+/obj/machinery/disposal/compactor/proc/compact()
+	if(stat & NOPOWER)
+		return
+	if(stat & BROKEN)
+		return
+	playsound(get_turf(src),'sound/machines/compactor.ogg', 30, 1) //Placeholder
+	flush = 1
+	flick("compactor_running",src)
+	spawn(41)
+		flush = 0
+		var/obj/item/trashcube/T =  new /obj/item/trashcube(get_turf(src))
+		for(var/obj/item/O in src)
+			flush_count = max(0, flush_count-3) //10% charge drained for each item
+			if(istype(O,/obj/item/trashcube) || O.w_class > W_CLASS_LARGE)
+				visible_message("<span class='warning'>\The [src] groans and spits out \the [O], prematurely ending the cycle.</span>")
+				launch(O)
+				flick("compactor_jobcancel",src)
+				break
+			O.forceMove(T)
+			T.w_class = max(T.w_class,O.w_class-1) //Make a cube starting at SMALL size, but increase it to medium if there is a large item inside.
+			T.update_icon()
+		if(emagged)
+			for(var/mob/M in src)
+				var/obj/item/weapon/reagent_containers/food/snacks/monkeycube/humancube/H = new(src)
+				H.contained_mob = M
+				M.forceMove(H)
+				launch(H)
+		if(!T.contents.len)
+			qdel(T) //If somehow we ended up with nothing inside
+
+/obj/machinery/disposal/compactor/proc/launch(var/atom/movable/AM)
```

Comment:
You know out of all the cosmetic stuff you pointed out this one actually would be worth changing if you know of a helper proc. So if you do know of one, please reply and I actually will PR to use that instead.

And for the record I actually do appreciate you bothering to look through for stuff (in spite of your veneer of sass that you insist on using for some inexplicable reason) and I would've changed some of this stuff if you had bothered to point it out in the *month* of sitting open for review.